generative/networks/nets/patchgan_discriminator.py

```diff
@@ -218,7 +218,10 @@ def __init__(
         )
 
         self.apply(self.initialise_weights)
-
+        if norm.lower() == 'batch' and torch.distributed.is_initialized():
+            print("WARNING: Discriminator is using BatchNorm and a distributed training environment has been detected. "
```

Comment:
```suggestion
            warnings.warn("WARNING: Discriminator is using BatchNorm and a distributed training environment has been detected. "
```

I suggest using warnings instead of printing to stdout so that they can be suppressed.